### ISSUE TITLE
Mac M1 install issue: TypeError: data type 'float128' not understood

### ISSUE THREAD

**Issue by Bea-Taylor** [pymc-devs/pymc#5691]
## Description of your problem
Installed on Mac M1 following instructions on GitHub: https://github.com/pymc-devs/pymc/wiki/Installation-Guide-(MacOS). 
When I import pyMC3, I get the following warnings: 

WARNING (theano.configdefaults): install mkl with `conda install mkl-service`: No module named 'mkl'
WARNING (theano.tensor.blas): Using NumPy C-API based implementation for BLAS functions.

**Please provide a minimal, self-contained, and reproducible example.**
```python
import pymc3 as pm 
```

**Please provide the full traceback.**

<details><summary>Complete error traceback</summary>

```python
TypeError                                 Traceback (most recent call last)
 Cell 1' in <cell line: 10>()
      [7](vscode-notebook-cell:/Users/bea/Documents/PhD/PPA_Staging/beallows/ranking_model-/ranking_model-/practise_with_pymall.ipynb#ch0000000?line=6)[ import mallows_kendall as mk 
      ]()[9](vscode-notebook-cell:/Users/bea/Documents/PhD/PPA_Staging/beallows/ranking_model-/ranking_model-/practise_with_pymall.ipynb#ch0000000?line=8)[ # Import pyMC
---> ]()[10](vscode-notebook-cell:/Users/bea/Documents/PhD/PPA_Staging/beallows/ranking_model-/ranking_model-/practise_with_pymall.ipynb#ch0000000?line=9)[ import pymc3 as pm

File /opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/__init__.py:41, in <module>
     ]()[38](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/__init__.py?line=37)[ __set_compiler_flags()
     ]()[40](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/__init__.py?line=39)[ from .blocking import *
---> ]()[41](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/__init__.py?line=40)[ from .distributions import *
     ]()[42](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/__init__.py?line=41)[ from .distributions import transforms
     ]()[43](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/__init__.py?line=42)[ from .glm import *

File /opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/__init__.py:15, in <module>
      ]()[1](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/__init__.py?line=0)[ #   Copyright 2020 The PyMC Developers
      ]()[2](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/__init__.py?line=1)[ #
      ]()[3](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/__init__.py?line=2)[ #   Licensed under the Apache License, Version 2.0 (the "License");
   (...)
     ]()[12](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/__init__.py?line=11)[ #   See the License for the specific language governing permissions and
     ]()[13](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/__init__.py?line=12)[ #   limitations under the License.
---> ]()[15](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/__init__.py?line=14)[ from . import timeseries
     ]()[16](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/__init__.py?line=15)[ from . import transforms
     ]()[17](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/__init__.py?line=16)[ from . import shape_utils

File /opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/timeseries.py:23, in <module>
     ]()[20](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/timeseries.py?line=19)[ import numpy as np
     ]()[22](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/timeseries.py?line=21)[ from pymc3.util import get_variable_name
---> ]()[23](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/timeseries.py?line=22)[ from .continuous import get_tau_sigma, Normal, Flat
     ]()[24](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/timeseries.py?line=23)[ from .shape_utils import to_tuple
     ]()[25](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/timeseries.py?line=24)[ from . import multivariate]()
[File /opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/continuous.py:32, in <module>
     ]()[30](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/continuous.py?line=29)[ from .special import log_i0
     ]()[31](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/continuous.py?line=30)[ from ..math import invlogit, logit, logdiffexp
---> ]()[32](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/continuous.py?line=31)[ from .dist_math import (
     ]()[33](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/continuous.py?line=32)[     alltrue_elemwise, betaln, bound, gammaln, i0e, incomplete_beta, logpow,
     ]()[34](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/continuous.py?line=33)[     normal_lccdf, normal_lcdf, SplineWrapper, std_cdf, zvalue,
     ]()[35](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/continuous.py?line=34)[     clipped_beta_rvs,
     ]()[36](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/continuous.py?line=35)[ )
     ]()[37](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/continuous.py?line=36)[ from .distribution import (Continuous, draw_values, generate_samples)
     ]()[39](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/continuous.py?line=38)[ __all__ = ['Uniform', 'Flat', 'HalfFlat', 'Normal', 'TruncatedNormal', 'Beta',
     ]()[40](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/continuous.py?line=39)[            'Kumaraswamy', 'Exponential', 'Laplace', 'StudentT', 'Cauchy',
     ]()[41](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/continuous.py?line=40)[            'HalfCauchy', 'Gamma', 'Weibull', 'HalfStudentT', 'Lognormal',
     ]()[42](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/continuous.py?line=41)[            'ChiSquared', 'HalfNormal', 'Wald', 'Pareto', 'InverseGamma',
     ]()[43](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/continuous.py?line=42)[            'ExGaussian', 'VonMises', 'SkewNormal', 'Triangular', 'Gumbel',
     ]()[44](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/continuous.py?line=43)[            'Logistic', 'LogitNormal', 'Interpolated', 'Rice', 'Moyal']

File /opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/dist_math.py:44, in <module>
     ]()[38](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/dist_math.py?line=37)[ _beta_clip_values = {
     ]()[39](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/dist_math.py?line=38)[     dtype: (np.nextafter(0, 1, dtype=dtype), np.nextafter(1, 0, dtype=dtype))
     ]()[40](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/dist_math.py?line=39)[     for dtype in ["float16", "float32", "float64"]
     ]()[41](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/dist_math.py?line=40)[ }
     ]()[42](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/dist_math.py?line=41)[ if platform.system() in ["Linux", "Darwin"]:
     ]()[43](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/dist_math.py?line=42)[     _beta_clip_values["float128"] = (
---> ]()[44](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/dist_math.py?line=43)[         np.nextafter(0, 1, dtype="float128"),
     ]()[45](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/dist_math.py?line=44)[         np.nextafter(1, 0, dtype="float128")
     ]()[46](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/dist_math.py?line=45)[     )
     ]()[50](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/dist_math.py?line=49)[ def bound(logp, *conditions, **kwargs):
     ]()[51](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/dist_math.py?line=50)[     """
     ]()[52](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/dist_math.py?line=51)[     Bounds a log probability density with several conditions.
     ]()[53](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/dist_math.py?line=52)[ 
   (...)
     ]()[67](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/dist_math.py?line=66)[     logp with elements set to -inf where any condition is False
     ]()[68](file:///opt/homebrew/Caskroom/miniforge/base/envs/pymc3_env/lib/python3.9/site-packages/pymc3/distributions/dist_math.py?line=67)[     """

TypeError: data type 'float128' not understood]()
```

</details>

**Please provide any additional information below.**
Read through https://discourse.pymc.io/t/install-headache-on-fresh-m1-macbook/7864 and https://discourse.pymc.io/t/pymc3-on-apple-m1-chip/6788/9 but didn't help. Tried installing from Git clone, but this also had similar errors. 

## Versions and main components

* PyMC/PyMC3 Version: 3.9.3
* Theano Version: 1.0.5
* Python Version: 3.9.12 
* Operating system: macOS Big Sur 11.4 
* How did you install PyMC/PyMC3: conda

**Comment by twiecki** [comment]
@Bea-Taylor This is something that's fixed in the upcoming 4.0 release. Are you able to upgrade to the beta?

**Comment by Bea-Taylor** [comment]
The beta fixes it- thank you! 😊
